lib/spack/spack/cmd/__init__.py

```diff
@@ -131,7 +133,8 @@ def parse_specs(args, **kwargs):
     tests = kwargs.get('tests', False)
 
     try:
-        specs = spack.spec.parse(args)
+        sargs = args if isinstance(args, six.string_types) else ' '.join(args)
+        specs = spack.spec.parse(sargs)
```

Comment:
Why's this change necessary?

Reply:
@becker It seems this change comes from f1fe707096f1b1ba4bc0de5a120c0b47dc13f0e8 you probably have a better understanding of this part to answer the question above.

Reply:
@becker33 I realize I misspelled your username above